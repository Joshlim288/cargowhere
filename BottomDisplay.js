import React, {useState} from 'react';
import {StyleSheet, View, Image, Pressable} from 'react-native';
import {
  NearbyScreen,
  FavouritesScreen,
  SearchScreen,
  AboutScreen,
} from './screens';

const tabNames = ['navigation', 'favourites', 'search', 'about'];

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  bar: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    height: '12%',
    borderBottomWidth: 1,
    borderColor: 'lightgrey',
  },
  activeTab: {
    borderBottomWidth: 1,
    borderBottomColor: '#2f60f4',
  },
  tab: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    width: 20,
    height: 20,
    tintColor: 'grey',
  },
  activeIcon: {
    tintColor: '#2f60f4',
  },
});

export default function NavigationBar(props) {
  const [activeTabs, setActiveTabs] = useState({
    navigation: true,
    favourites: false,
    search: false,
    about: false,
  });

  // set styles for active/inactive tabs
  const activeTabStyle = [styles.tab, styles.activeTab];
  const inactiveTabStyle = styles.tab;
  const activeIconStyle = [styles.icon, styles.activeIcon];
  const inactiveIconStyle = styles.icon;

  return (
    <View style={styles.container}>
      <View style={styles.bar}>
        <Pressable
          style={activeTabs.navigation ? activeTabStyle : inactiveTabStyle}
          onPress={() =>
            setActiveTabs({
              navigation: true,
              favourites: false,
              search: false,
              about: false,
            })
          }
          android_ripple={{color: 'lightgrey'}}>
          <Image
            style={activeTabs.navigation ? activeIconStyle : inactiveIconStyle}
            source={require('./images/navigation_arrow.png')}
          />
        </Pressable>

        <Pressable
          style={activeTabs.favourites ? activeTabStyle : inactiveTabStyle}
          onPress={() =>
            setActiveTabs({
              navigation: false,
              favourites: true,
              search: false,
              about: false,
            })
          }
          android_ripple={{color: 'lightgrey'}}>
          <Image
            style={activeTabs.favourites ? activeIconStyle : inactiveIconStyle}
            source={require('./images/heart.png')}
          />
        </Pressable>

        <Pressable
          style={activeTabs.search ? activeTabStyle : inactiveTabStyle}
          onPress={() =>
            setActiveTabs({
              navigation: false,
              favourites: false,
              search: true,
              about: false,
            })
          }
          android_ripple={{color: 'lightgrey'}}>
          <Image
            style={activeTabs.search ? activeIconStyle : inactiveIconStyle}
            source={require('./images/search.png')}
          />
        </Pressable>

        <Pressable
          style={activeTabs.about ? activeTabStyle : inactiveTabStyle}
          onPress={() =>
            setActiveTabs({
              navigation: false,
              favourites: false,
              search: false,
              about: true,
            })
          }
          android_ripple={{color: 'lightgrey'}}>
          <Image
            style={activeTabs.about ? activeIconStyle : inactiveIconStyle}
            source={require('./images/information.png')}
          />
        </Pressable>
      </View>

      {/* Switching of tab screen is done here */}
      {activeTabs.navigation && <NearbyScreen  />}
      {activeTabs.favourites && <FavouritesScreen  />}
      {activeTabs.search && 
        <SearchScreen 
          setRegion={props.setRegion}
          setSpecificLocation={props.setSpecificLocation}
          carparks = {props.carparks}
        />
      }
      {activeTabs.about && <AboutScreen />}
    </View>
  );
}
