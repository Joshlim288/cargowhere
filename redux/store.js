import {configureStore} from '@reduxjs/toolkit';

import carparksReducer from './carparksSlice';
import regionReducer from './regionSlice';
import specificLocationReducer from './specificLocationSlice';
import sortCriteriaReducer from './sortCriteriaSlice';
import availabilityReducer from './availabilitySlice';
import maxCarparksReducer from './maxCarparksSlice';
import selectedFavouriteReducer from './selectedFavouriteSlice';
import alertReducer from './alertSlice';

// our store will have the states carparks and region, and all updates to these states will be done with their reducers
export default configureStore({
  reducer: {
    carparks: carparksReducer,
    region: regionReducer,
    specificLocation: specificLocationReducer,
    sortCriteria: sortCriteriaReducer,
    availability: availabilityReducer,
    maxCarparks: maxCarparksReducer,
    selectedFavourite: selectedFavouriteReducer,
    alert: alertReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false,
    }),
});
