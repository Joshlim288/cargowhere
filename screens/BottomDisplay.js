import React, {useState} from 'react';
import {StyleSheet, View, Image, Pressable} from 'react-native';
import {
  NearbyScreen,
  FavouritesScreen,
  AboutScreen,
} from '../screens';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  bar: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    height: '12%',
    borderBottomWidth: 1,
    borderColor: 'lightgrey',
  },
  activeTab: {
    borderBottomWidth: 1,
    borderBottomColor: '#2f60f4',
  },
  tab: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    width: 20,
    height: 20,
    tintColor: 'grey',
  },
  activeIcon: {
    tintColor: '#2f60f4',
  },
});

export default function NavigationBar(props) {
  const [activeTabs, setActiveTabs] = useState({
    navigation: true,
    favourites: false,
    about: false,
  });

  // set styles for active/inactive tabs
  const activeTabStyle = [styles.tab, styles.activeTab];
  const inactiveTabStyle = styles.tab;
  const activeIconStyle = [styles.icon, styles.activeIcon];
  const inactiveIconStyle = styles.icon;

  return (
    <View style={styles.container}>
      {/* display tab menu bar */}
      <View style={styles.bar}>
        {/* NearbyScreen tab */}
        <Pressable
          style={activeTabs.navigation ? activeTabStyle : inactiveTabStyle}
          onPress={() =>
            setActiveTabs({
              navigation: true,
              favourites: false,
              about: false,
            })
          }
          android_ripple={{color: 'lightgrey'}}>
          <Image
            style={activeTabs.navigation ? activeIconStyle : inactiveIconStyle}
            source={require('../images/navigation_arrow.png')}
          />
        </Pressable>

        {/* Favourites tab */}
        <Pressable
          style={activeTabs.favourites ? activeTabStyle : inactiveTabStyle}
          onPress={() =>
            setActiveTabs({
              navigation: false,
              favourites: true,
              about: false,
            })
          }
          android_ripple={{color: 'lightgrey'}}>
          <Image
            style={activeTabs.favourites ? activeIconStyle : inactiveIconStyle}
            source={require('../images/heart.png')}
          />
        </Pressable>

        {/* About tab */}
        <Pressable
          style={activeTabs.about ? activeTabStyle : inactiveTabStyle}
          onPress={() =>
            setActiveTabs({
              navigation: false,
              favourites: false,
              about: true,
            })
          }
          android_ripple={{color: 'lightgrey'}}>
          <Image
            style={activeTabs.about ? activeIconStyle : inactiveIconStyle}
            source={require('../images/information.png')}
          />
        </Pressable>
      </View>

      {/* Switching of tab screen is done here */}
      {activeTabs.navigation && (
        <NearbyScreen
          sortCriteriaChanged={props.sortCriteriaChanged}
        />
      )}
      {activeTabs.favourites && (
        <FavouritesScreen/>
      )}
      {activeTabs.about && <AboutScreen />}
    </View>
  );
}
