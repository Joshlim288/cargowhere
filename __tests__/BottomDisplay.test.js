/**
 * @format
 */
import React from 'react';
import BottomDisplay from '../BottomDisplay';

import {Provider} from 'react-redux';
import store from '../redux/store';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

/*
 let outputData = '';
 var storeLog = (inputs) => (outputData += inputs);
 console['log'] = jest.fn(storeLog);
 console['error'] = jest.fn(storeLog);
 These lines of code are so that Jest doesn't throw an error 
 when the components attempt to console.log() or console.error(),
 especially when there are async functions that will console.log or console.error
 after Jest is done testing that it renders correctly
 */

it('BottomDisplay renders correctly', () => {
  let outputData = '';
  var storeLog = (inputs) => (outputData += inputs);
  console.log = jest.fn(storeLog);
  console.error = jest.fn(storeLog);
  const tree = renderer
    .create(
      <Provider store={store}>
        <BottomDisplay />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
