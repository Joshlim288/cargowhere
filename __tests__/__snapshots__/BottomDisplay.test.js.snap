// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomDisplay renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgrey",
        "flexDirection": "row",
        "height": "12%",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "borderBottomColor": "#2f60f4",
            "borderBottomWidth": 1,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/navigation_arrow.png",
          }
        }
        style={
          Array [
            Object {
              "height": 20,
              "tintColor": "grey",
              "width": 20,
            },
            Object {
              "tintColor": "#2f60f4",
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/heart.png",
          }
        }
        style={
          Object {
            "height": 20,
            "tintColor": "grey",
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/information.png",
          }
        }
        style={
          Object {
            "height": 20,
            "tintColor": "grey",
            "width": 20,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": "100%",
          "left": "0%",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#eee",
            "borderBottomColor": "#eee",
            "borderBottomWidth": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "paddingLeft": 20,
              "paddingTop": 10,
              "textDecorationLine": "underline",
            }
          }
        >
          Nearby
        </Text>
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 20,
              "paddingTop": 10,
              "textAlign": "right",
            }
          }
        >
          Sort By:
        </Text>
        <View
          style={
            Object {
              "color": "black",
              "width": "43%",
            }
          }
        >
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Distance",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "distance",
                },
                Object {
                  "label": "Availability",
                  "testID": undefined,
                  "textColor": undefined,
                  "value": "availability",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "height": "100%",
          "left": "100%",
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
